final/background/spoofax.tex

```diff
@@ -87,211 +119,139 @@ \subsubsection{SDF3: syntax definition in Spoofax}
-specified in Java instead. Therefore the interface is bidirectional:
-Stratego can hook into Java, and Java can use the Stratego API.
+Stratego terms can however represent any tree structure, not just ASTs. As such,
+when using Spoofax programmatically through its API, one sees Stratego terms
+used as an internal representation of the data going through the execution
+pipeline depicted in \cref{fig:spoofax-relations}.
 
 \subsection{Dynamic Semantics}
 \label{ssec:dynamic-semantics}
 Dynamic semantics refers to how a program written in some language
-behaves~\cite{Winskel93}. There are many approaches to formally
-specify the dynamic semantics of a programming language (for an
-extensive treatment, see~\cite{Winskel93}). For this section only
-one sort of approach called \emph{operational semantics} is relevant.
+behaves~\cite{Winskel93}. There are many approaches to formally specify the
+dynamic semantics of a programming language (for an extensive treatment,
+see~\cite{Winskel93}). One way, for example, is to specify the transformation of
+programs written in the programming language to programs of another language of
+which the dynamic semantics are already known.
+
+DynSem~\cite{VerguNV15}, a DSL for the specification of dynamic semantics, uses
+an approach called \textit{operational semantics}. The rest of this section
+discusses how DynSem (and operational semantics) is used to specify the dynamic
+semantics of a language.
 
 \subsubsection{DynSem: rule-based dynamic semantics}
 \label{ssec:dynsem}
-Aside from Stratego, the Spoofax team has developed an additional
-method to declare the dynamic semantics of a language, namely a DSL
-called \emph{DynSem}~\cite{VerguNV15}. DynSem allows for an operational
-semantics specification from which a Java-based AST interpreter is
-automatically generated.
-
-In DynSem, like other meta-languages in Spoofax, the dynamic semantics
-are specified by means of syntax-directed rules. To show how rules can
-define the dynamic semantics of a language, consider the classic
-example of the \(\beta\)-reduction, which defines function application in
-the lambda calculus. The rule replaces all the occurences of the
-parameter \(x\) with the argument \(e_2\), within the expression \(e_1\):
-
-\begin{equation}
-(\lambda x.e_1) e_2 \rightarrow e_1[x := e_2]
-\end{equation}
-
-In a similar way, dynamic semantics can be specified in DynSem using a
-syntax very similar to the formal syntax used in the literature. Take
-here the example of defining the behaviour of some boolean operators
-in paplj:
-\lstset{language=dynsem,numbers=left}
-\begin{lstlisting}
+Aside from Stratego, the Spoofax team has developed an additional method to
+declare the dynamic semantics of a language, namely a DSL called
+DynSem~\cite{VerguNV15}. DynSem allows for an operational semantics
+specification from which a Java-based AST interpreter can be automatically
+derived.
```

Comment:
I keep this introduction and remove the one in the paragraph that comes before it.